### ISSUE TITLE
Unable to Compile as Crate

### ISSUE THREAD

**Issue by AlexCreamer** [bluss/petgraph#157]
Hi, I'm using Rust nightly (1.20) and I'm trying to compile petgraph since it's a dependency of one of my packages and I get the following errors:
```
error[E0034]: multiple applicable items in scope
   --> /home/onix/.cargo/registry/src/github.com-1ecc6299db9ec823/petgraph-0.2.9/src/graph.rs:421:17
    |
421 |         assert!(Ix::max().index() == !0 || NodeIndex::end() != node_idx);
    |                 ^^^^^^^ multiple `max` found
    |
note: candidate #1 is defined in the trait `graph::IndexType`
   --> /home/onix/.cargo/registry/src/github.com-1ecc6299db9ec823/petgraph-0.2.9/src/graph.rs:31:5
    |
31  |     fn max() -> Self;
    |     ^^^^^^^^^^^^^^^^^
    = help: to disambiguate the method call, write `graph::IndexType::max(...)` instead
note: candidate #2 is defined in the trait `std::cmp::Ord`
    = help: to disambiguate the method call, write `std::cmp::Ord::max(...)` instead

error[E0034]: multiple applicable items in scope
   --> /home/onix/.cargo/registry/src/github.com-1ecc6299db9ec823/petgraph-0.2.9/src/graph.rs:458:17
    |
458 |         assert!(Ix::max().index() == !0 || EdgeIndex::end() != edge_idx);
    |                 ^^^^^^^ multiple `max` found
    |
note: candidate #1 is defined in the trait `graph::IndexType`
   --> /home/onix/.cargo/registry/src/github.com-1ecc6299db9ec823/petgraph-0.2.9/src/graph.rs:31:5
    |
31  |     fn max() -> Self;
    |     ^^^^^^^^^^^^^^^^^
    = help: to disambiguate the method call, write `graph::IndexType::max(...)` instead
note: candidate #2 is defined in the trait `std::cmp::Ord`
    = help: to disambiguate the method call, write `std::cmp::Ord::max(...)` instead

   Compiling tempfile v2.1.5
error: aborting due to 2 previous errors

error: Could not compile `petgraph`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by messense** [comment]
Duplicated of #153 , resolved by #154 , you should upgrade `petgraph` to 0.4.5.

**Comment by bluss** [comment]
Hi, thanks for the report. See #154. `cargo update` can have cargo get the latest version (matching the version spec like "0.4") for you, but the best solution is to put the version in Cargo.toml to 0.4.5, since that solves the issue regardless of Cargo.lock files lying around.
